exporter/elasticsearchexporter/exporter.go

```diff
@@ -145,8 +144,16 @@ func (e *elasticsearchExporter) pushEvent(ctx context.Context, document []byte)
 				zap.NamedError("reason", err))
 
 			attempts++
-			body.Seek(0, io.SeekStart)
-			e.bulkIndexer.Add(ctx, item)
+			_, seekerr := body.Seek(0, io.SeekStart)
+			if seekerr != nil {
+				e.logger.Error("failed to set the next offset.",
+					zap.NamedError("reason", seekerr))
+			}
+			indexererr := e.bulkIndexer.Add(ctx, item)
+			if indexererr != nil {
```

Comment:
I prefer this (and everywhere else) to inline this:

```golang
if indexererr := e.bulkIndexer.Add(ctx, item); if indexererr != nil {
   ...
}
```

Reply:
have done this will push as all improvements suggested are finalised